packages/react/src/components/Dialog/index.tsx

```diff
@@ -119,8 +127,6 @@ export default class Dialog extends React.Component<DialogProps, DialogState> {
       <FocusTrap
         focusTrapOptions={{
           allowOutsideClick: true,
-          onDeactivate: this.close,
-          escapeDeactivates: !forceAction,
```

Comment:
Looking at [`focus-trap` options](https://github.com/focus-trap/focus-trap#createoptions) it looks like we're not correctly handling this yet. `escapeDeactivates` defaults to `true`, so we likely want to always set this to `false`.